src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -1019,6 +1019,13 @@ public string[] LiteralPath
 
         private bool _isLiteralPath;
 
+        /// <summary>
+        /// Gets or sets a value indicating if only string values containing matched lines should be returned.
+        /// If not (default) return MatchInfo (or bool objects, when Quiet is passed).
+        /// </summary>
+        [Parameter]
+        public SwitchParameter Raw { get; set; }
```

Comment:
This should probably be in a separate parameter set to the `Quiet` switch, otherwise behaviour when both are supplied will be confusing regardless of which is provided.

They both transform the output, so we shouldn't allow them to be present at the same time imo. This would also let you simplify your logic further down a bit by checking `ParameterSetName` before looking at Raw/Quiet

Reply:
Thanks for spotting that, I've thought about this. Did you see my comment here: 

> The approach I've taken does not introduce new parameter sets, even though -Raw is not compatible with -Quiet and -Context. In the case -Raw is used together with -Context, the latter is ignored. To me, it would not make sense to have context lines when all we want is matched lines. In the case -Raw is used together with -Quiet, Quiet takes precedence, because the choice for precedence seemed somewhat arbitrary and it was easier to implement this way. One might argue that respecting a users whish for -Quiet behaviour is more important than returning strings in that case.

The -List and -Quiet params did not go in separate parameter sets either. That also made me swing this way.

Reply:
I hear you, and maybe the PS team will have a different opinion, but in my mind it's a lot better from a UX standpoint to have these separated out in different sets. 

It's too easy to get confused as to which output takes precedence. We have the ability to separate them out, so I'd say... why not? 😄

Reply:
OK, no worries. I'll put -Raw and -Quiet in different paramsets then 😉 . First thing tomorrow morning!

Reply:
I would just say that don't assume the current code is the best way to do something :)  PowerShell is a large code base written by a large number of people over a large number of years.  It's generally rare that old code is updated when new and better patterns emerge.  In this specific case, my preference is to have different parameter sets and consider fixing `-List` and `-Quiet` (unless there's a clear reason why it was done this way, but doesn't have to be part of this PR, can be separate one).  Note that we do have specific cases where we decided it was best to not have parametersets and do a runtime check due to the complexity of many parameters that were mutually exclusive over several existing parametersets, but those should be exceptions.

Reply:
@SteveL-MSFT I think `Where-Object` begs to differ. I've seen those very very naughty parameter sets 😉

Reply:
param sets it is!

Reply:
@SteveL-MSFT I've implemented the requested changes. CI is happy now. I'm not sure how to mark your requested change as 'done' :)